options.go

```diff
@@ -70,10 +82,26 @@ func EventRegexMustCompile() Option {
 
 type eventRegexMustCompileOption struct{}
 
-func (eventRegexMustCompileOption) Validate(val Validator) error {
-	if err := val.ValidateEventRegex(); err != nil {
-		return err
+func (eventRegexMustCompileOption) Validate(i any) error {
+	switch r := i.(type) {
+	case *RegistrationV1:
+		for _, e := range r.Events {
+			_, err := regexp.Compile(e)
+			if err != nil {
+				return fmt.Errorf("%w: unable to compile matching", ErrInvalidInput)
+			}
+		}
+	case *RegistrationV2:
+		for _, m := range r.Matcher {
+			_, err := regexp.Compile(m.Regex)
+			if err != nil {
+				return fmt.Errorf("%w: unable to compile matching", ErrInvalidInput)
+			}
+		}
+	default:
+		return fmt.Errorf("%w: Registration must be of type RegistrationV1 or RegistrationV2", ErrInvalidType)
```

Comment:
I would make `Registration must be of type RegistrationV1 or RegistrationV2` it's own error and then use `errors.Join` to group the two errors

Reply:
Using ErrUnkownType for default